flow/core/kernel/scenario/traci.py

```diff
@@ -0,0 +1,885 @@
+
+        # name of the .net.xml file (located in cfg_path)
+        self.netfn = netfn
+
+        # collect data from the generated network configuration file
+        edges_dict, conn_dict = self._import_edges_from_net()
+
+        return edges_dict, conn_dict
+
+    def generate_net_from_netfile(self, net_params):
+        """Pass relevant data from an already processed .net.xml file.
+
+        This method is used to collect the edges and connection data from a
+        netfile and pass it to the scenario class for later use.
+
+        Parameters
+        ----------
+        net_params : flow.core.params.NetParams type
+            network-specific parameters. Different networks require different
+            net_params; see the separate sub-classes for more information.
+
+        Returns
+        -------
+        edges : dict <dict>
+            Key = name of the edge
+            Elements = length, lanes, speed
+        connection_data : dict < dict < list<tup> > >
+            Key = name of the arriving edge
+                Key = lane index
+                Element = list of edge/lane pairs that a vehicle can traverse
+                from the arriving edge/lane pairs
+        """
+        # name of the .net.xml file (located in cfg_path)
+        self.netfn = net_params.netfile
+
+        # collect data from the generated network configuration file
+        edges_dict, conn_dict = self._import_edges_from_net()
+
+        return edges_dict, conn_dict
+
+    def generate_cfg(self, net_params, traffic_lights, routes):
+        """Generate .sumo.cfg files using net files and netconvert.
+
+        This includes files such as the routes vehicles can traverse,
+        properties of the traffic lights, and the view settings of the gui
+        (whether the gui is used or not). The background of the gui is set here
+        to be grey, with RGB values: (100, 100, 100).
+
+        Parameters
+        ----------
+        net_params : NetParams type
+            see flow/core/params.py
+        traffic_lights : flow.core.traffic_lights.TrafficLights type
+            traffic light information, used to determine which nodes are
+            treated as traffic lights
+        routes : dict
+            Key = name of the starting edge
+            Element = list of edges a vehicle starting from this edge must
+            traverse.
+        """
+        # specify routes vehicles can take
+        self.rts = routes
+
+        add = makexml('additional',
+                      'http://sumo.dlr.de/xsd/additional_file.xsd')
+
+        # add the routes to the .add.xml file
+        for (edge, route) in self.rts.items():
+            add.append(E('route', id='route%s' % edge, edges=' '.join(route)))
+
+        # add (optionally) the traffic light properties to the .add.xml file
+        num_traffic_lights = len(list(traffic_lights.get_properties().keys()))
+        if num_traffic_lights > 0:
+            if traffic_lights.baseline:
```

Comment:
we should rename this; in one case it is called baseline and in the other "actuated_default"; it's confusing

Reply:
ya, I don't know anything about these.. there's an issue about writing tests for these. I think if we do that we will begin to result this, something I'm currently talking to Yashar about